config/main.py

```diff
@@ -2064,6 +2064,538 @@ def delete_snmptrap_server(ctx, ver):
+        sys.exit(3)
+    else:
+        db.cfgdb.set_entry('SNMP_COMMUNITY', community, {'TYPE': string_type})
+        click.echo("SNMP community {} added to configuration".format(community))
+        try:
+            click.echo("Restarting SNMP service...")
+            clicommon.run_command("systemctl reset-failed snmp.service", display_cmd=False)
+            clicommon.run_command("systemctl restart snmp.service", display_cmd=False)
+        except SystemExit as e:
+            db.cfgdb.set_entry('SNMP_COMMUNITY', community, None)
+            click.echo("Restart service snmp failed with error {}".format(e))
+            raise click.Abort()
+
+
+@community.command('del')
+@click.argument('community', metavar='<snmp_community>', required=True)
+@clicommon.pass_db
+def del_community(db, community):
+    """ Delete snmp community string"""
+    snmp_communities = db.cfgdb.get_table("SNMP_COMMUNITY")
+    if community not in snmp_communities:
+        click.echo("SNMP community {} is not configured".format(community))
+        sys.exit(1)
+    else:
+        db.cfgdb.set_entry('SNMP_COMMUNITY', community, None)
+        click.echo("SNMP community {} removed from configuration".format(community))
+        try:
+            click.echo("Restarting SNMP service...")
+            clicommon.run_command("systemctl reset-failed snmp.service", display_cmd=False)
+            clicommon.run_command("systemctl restart snmp.service", display_cmd=False)
+        except SystemExit as e:
+            click.echo("Restart service snmp failed with error {}".format(e))
+            raise click.Abort()
+
+
+@community.command('replace')
+@click.argument('current_community', metavar='<current_community_string>', required=True)
+@click.argument('new_community', metavar='<new_community_string>', required=True)
+@clicommon.pass_db
+def replace_community(db, current_community, new_community):
+    """ Replace snmp community string"""
+    snmp_communities = db.cfgdb.get_table("SNMP_COMMUNITY")
+    if not current_community in snmp_communities:
+        click.echo("Current SNMP community {} is not configured".format(current_community))
+        sys.exit(1)
+    if not snmp_community_secret_check(new_community):
+        click.echo("Bad community")
+        sys.exit(2)
+    elif new_community in snmp_communities:
+        click.echo("New SNMP community {} to replace current SNMP community {} already "
+                   "configured".format(new_community, current_community))
+        sys.exit(3)
+    else:
+        string_type = snmp_communities[current_community]['TYPE']
+        db.cfgdb.set_entry('SNMP_COMMUNITY', new_community, {'TYPE': string_type})
+        click.echo("SNMP community {} added to configuration".format(new_community))
+        db.cfgdb.set_entry('SNMP_COMMUNITY', current_community, None)
+        click.echo('SNMP community {} removed from configuration'.format(current_community))
```

Comment:
>click.echo('SNMP community {} removed from configuration'.format(current_community)) [](http://example.com/codeflow?start=8&length=84)

Too verbose, you can remove this line and the next line is good enough.

Reply:
Sure I'm removing that.